versioning/versioning.md

```diff
@@ -0,0 +1,113 @@
+
+* What version matrices does a Mobile App Developer need to understand the system? 
+* At what level are versions kept?
+* When changes are made to different components, what is the impact on dependent components?
+* How do the QE team know what to test, with respect to what version of what component should work together?
+ 
+
+## Asumptions
+Listing of assumptions on how other parts of the system will work, as the proposal may base some decisions on these assumptions.
+
+* An assumption is that the OpenShift Catalog will, in the future, publish versions associated with each Service. It won't act on them, just display the version.
+
+  
+## Proposed Solution
+A Mobile Service (e.g. Synch, Push, Auth) is made up of a number of sub-components, namely;
+
+* Mobile Service
+  * Backend Component
+  * Mobile Service APB
+  * Mobile Client Service SDK (Android)
+  * Mobile Client Service SDK (iOS)
+  * Mobile Client Service SDK (Cordova)
+  * UI - Service may or may not have a UI component
+  * CLI - Service may or may not have a specific aspect of the CLI dedicated to its functionality
+  * IDE Plugins - Service may or may not have a specific aspect of an IDE Plugin dedicated to its functionality (TBD)
+    * Android Studio
+    * VS Code
+    * XCode
+  * E2E Service Tests - suite of tests for the version of the Mobile Service
+
+All of the sub-components above will have their own version.
+
+### Introduce an overall **Mobile Service version**
+This version is like a snapshot in time of everything (all specific sub-components) which makes up that Mobile Service. See table below for overall version of Synch 1.3.
+
+|               | Synch 1.3        |
+|  -----------  |  --------------  |
+| APB                  | v1.0 |
+| Backend Serice       | v1.2 |
+| SDK-Android          | v1.1 |
+| SDK-iOS              | v1.0 |
+| SDK-Cordova          | v1.0 |
+| IDE-AndroidStudio    | v1.2 |
+| IDE-VSCode           | v1.0 |
+| IDE-XCode            | v1.0 |
+| UI                   | v1.0 |
+| CLI                  | v1.0 |
+| E2E Service Tests    | v1.0 |
+
+The reason for introducing this is two fold;
+* to encapsulate all of the sub components and their versions such that developers can easily identify what versions of the sub-components make up that version of the service and therefore work together.
+* to have some high level entity which has a version when considering Mobile Service integrations (i.e. what Mobile Services can work together and what versions of these services work together)
+
+This table of each overall version of the Mobile Serivce and its set of sub-components must be produced and made availble through the docs. This table of information is essential for developers.
+
```

Comment:
IHMO the user should not care about the tags when run/install/setup the mobile service. The latest version should be the latest released one. 

Regards the internal versions(deps) the user should be able to check the versions used to build their projects, I mean what libs/dep/version each project/application/service is using by each component in the case be able to troubleshoot it. For example, if the customer is facing a bug when trying login via the client app which is using the keylock feature they should be able via the OCP get the version of keylock feature used and tell us which will allow we check if that bug is solved or not in future version in the same way that I understand which it is possible to get done with the SDKs already.

Reply:
> Then, this matrix should be not required at all.

It will be required for SDK/Showcase unless we are going to include them in UI (if we do then we can avoid entire proposal and complexity)